gax-java/gax/src/main/java/com/google/api/gax/rpc/Callables.java

```diff
@@ -115,7 +115,8 @@ public static <RequestT, ResponseT> ServerStreamingCallable<RequestT, ResponseT>
         callable.withDefaultCallContext(
             clientContext
                 .getDefaultCallContext()
-                .withStreamIdleTimeout(callSettings.getIdleTimeout()));
+                .withStreamIdleTimeout(callSettings.getIdleTimeout())
+                .withStreamWaitTimeout(callSettings.getWaitTimeout()));
```

Comment:
Do you know what the behavior is when either (or both) the idleTimeout and waitTimeout values are greater than the Watchdog's scheduleInterval value? I'm guessing that the Watchdog does nothing and makes me wonder if there should (or needs to) be a check to ensure that the scheduleInterval is less than both of these timeout values (when the values are non-zero).

Kinda like how we have a check if RPC timeout is greater than total timeout: `java.lang.IllegalStateException: max rpc timeout must not be shorter than initial timeout`

Reply:
I think watchdog will kill the stream anyways whenever it sees the idleTimeout and waitTimeout are expired. So I don't feel it's necessary to compare timeouts with checkInterval here?

Reply:
Also the default checkInterval is pretty small (10 seconds). So I don't think it'll be a problem.

Reply:
Actually, on a second look .. I think may be misunderstanding how Watchdog works. I think this is fine.